modules/lang-mustache/src/main/java/org/elasticsearch/script/mustache/RestSearchTemplateAction.java

```diff
@@ -76,7 +77,7 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
             searchTemplateRequest = SearchTemplateRequest.fromXContent(parser);
         }
         searchTemplateRequest.setRequest(searchRequest);
-
+        CCSVersionCheckHelper.checkCCSVersionCompatibility(request, searchTemplateRequest);
```

Comment:
I don't see why those two APIs need to be addressed as a follow-up. I would simply expose the flag in every API that supports CCS.

Reply:
I added "_terms_enum" and "_resolve". On closer inspection I'm not really sure we need checks on "_resolve" since at least for now there is little inside the request that is version dependent, so its hard to write tests for it for example. Not sure how we want to make sure we don't add any stuff that breaks N-1 compatibiity in the future then though. Opinions?

Reply:
I think the point is to have the same flag exposed on any API that Kibana may call that supports remote clusters, that is why I thought it may make sense to be consistent.